### PR TITLE
Change switch expression to a new precedence level

### LINKED ISSUE #34472: Change switch expression to a new precedence level
On 2019-03-13 the LDM decided to change the precedence of the switch expression to a new precedence level between unary and multiplicative.  This issue tracks the implementation of that decision.

### PR DESCRIPTION
**Pull request by gafter**
Fixes #34472

### CODE REVIEW

File: src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs [hunk 1]
```diff
@@ -7442,6 +7441,234 @@ public void SwitchExpressionPrecedence_02()
...
+                N(SyntaxKind.CloseBraceToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void SwitchExpressionPrecedence_12()
+        {
+            UsingExpression("a switch {}!",
+                // (1,1): error CS1073: Unexpected token '!'
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs]: So if I need to suppress the switch expression, I need to wrap it in parens?

Comment (gafter) [reply src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs]: Yes, exactly.

**Comment by CyrusNajmabadi** [comment]
@gafter Can you open an IDE bug on this change?  There are several IDE features that care about precedence (for example, if we offer parentheses or attempt to remove them).  THe addition of an entirely new precedence level means IDE should def at least audit those features to make sure there's no unintended fallout here.  I don't anticipate any problems, but it would be good to have someone take a look. 

Thanks!

**Comment by CyrusNajmabadi** [comment]
Tagging @sharwell for awareness on https://github.com/dotnet/roslyn/pull/34481#issuecomment-477234403

**Pull request merged**